### DISCUSSION TITLE
Mail scanner - options for improvement

### DISCUSSION THREAD

**Discussion by TonyM15** [YetiForceCompany/YetiForceCRM#15170] [Feature improvement]
Mail scanners out of the box appears to be in need of some options where essential would be separating emails fields, so scanner can recognize following fields: FROM, TO, CC, BCC, SUBJECT content, BODY content.
Currently if user is in CC  - for instance - scanner creating separate ticket, in fact misleading CC user on "real" addressee of email/ticket (especially when email is not started with "Dear xxx". If we can determine set of fields in email and transfer this data to ticket itself, it would be much easier to deal with assignment/statuses with use of workflows.
What do you gents think?

**Comment by waran70** [comment]
Interesting point unless there is some hidden (out of programming scope) options.

**Reply by waran70** [reply]
There would be way around (still too complex I think) if Email History can be mapped with other modules (currently it cannot).
If so, ticket could be created from one mailbox and shared with CC,BCC user (with some custom status) plus some WF condition's could be adjusted. Mapping would be the fastest way to achieve author needs

**Reply by garryatbrighton** [reply]
Interesting... Waran70 is very close... 
In workflow we can create custom mapping BUT cannot assign it to current RC_user.
Can developer point out (and add to core) how to add "rc_user" to the list of "Assigned to" and "Shared with users" fields in WF conditions?

![image](https://user-images.githubusercontent.com/80203589/132996115-5075d6b8-aa87-4dfc-b294-368d1747e7c5.png)

**Reply by blu-IT** [reply]
Good idea!

Personally I have given up to handle tickets / customer requests with YF and switched to Zammd (OTRS fork). I created serveral issues in Github, because of the strange behavior of e-mail handling in YF. E.g when responding to a mail/ticket from the customer in YF and the customer responds to the mail again, YF always creates an **NEW** ticket. Also ticket no. is not displayed in the subject etc. so I switched to a system that can deal with such basics out of the box.

**Comment by Marshal4582** [comment]
Tried to use it for professional Customer service, but too many obstacles... indeed, if we talking about professional CRM this part needs urgent attention

**Comment by s2b-git** [comment]
My wish-list :-)

1. Allow convert email to comment. Currently when user comment on ticket, the email is sent but reply is attached to ticket as an email and not converted to comment. This breaking conversation thread order. Comments are perfect place to track what was said in relation to current ticket.
2. Add possibility to attach files to comment
3. Add to mailscanner configurable conditions:
- from: equal to | not equal to | contain | not contain
- to: equal to | not equal to | contain | not contain
- cc: equal to | not equal to | contain | not contain
- subject:  contains | not contains
- body:  contains | not contains
Conditions can be done with REGEX but having clear filters would beautify experience
4. Add cascade owner changing (the same concerning Shared with users: field I think) including related emails (currently read only)

Best

**Reply by s2b-git** [reply]
I want to add that mail scanner configuration doesn't allow for autoresponder what is pretty common needs here and there :)

**Reply by s2b-git** [reply]
Need to admit that I was wrong. Selecting email address from related Emails list (workflow) allowing for auto-responding.
Learning yetiforce is often positively challenging task.

Best

**Reply by diderich** [reply]
I don't think that converting e-mails to comments is the right solution to your problem. What you are looking for is a reporting timeline, like in the picture below that regroups elements from different modules and displays them sorted along the timeline. Unfortunately, such a functionality currently does not exist. I developed my own local version of it as shown in the figure (which is based on a highly customized version of YetiForceCRM, i.e., additional fields, different naming of elements, different semantics for leads, etc.). It is really not that complicated to develop, but it is a tedious task, as each database table looks different and uses different naming conventions. If anyone wants to take up the task to develop such a functionality to be included in the YetiForceCRM distribution, I am more than happy to share my code (without any support of warranty, unfortunately, under an MIT license).

![reporting png](https://user-images.githubusercontent.com/10230144/191291222-7b5a65e8-e0e1-4e43-924a-87ff5b5b8c02.jpg)
